cvetools/cvesearch.go

```diff
@@ -163,21 +163,22 @@ func (cv *ScanTools) ScanImageData(data *share.ScanData) (*share.ScanResult, err
 
 	appPkgs := scan.NewScanApps(true).DerivePkg(pkgs)
 	if k8sAppString != "" {
+		log.WithFields(log.Fields{"k8sAppRes": k8sAppString}).Debug()
 		// append k8s repo into the appPkgs
 		if i := strings.Index(k8sAppString, ":"); i > 0 {
 			// remove the RKE variants
-			name := strings.TrimPrefix(k8sAppString[:i], "rke2-")
-			name = strings.TrimPrefix(name, "rke-")
-			name = fmt.Sprintf("go:k8s.io/%s", name)
-			version := strings.TrimPrefix(k8sAppString[i+1:], "v")
-			k8sAppRes := scan.AppPackage{
-				AppName:    "golang",        // belong to golang type
-				ModuleName: name,            // image name
-				Version:    version,         // tag
-				FileName:   scan.Kubernetes, // high-level k8s scan
+			name := strings.TrimPrefix(strings.TrimPrefix(k8sAppString[:i], "rke2-"), "rke-")
+			if name == "ingress-nginx" { // should expand into a broader range later
+				name = fmt.Sprintf("go:k8s.io/%s", name)
+				version := strings.TrimPrefix(k8sAppString[i+1:], "v")
+				k8sAppRes := scan.AppPackage{
+					AppName:    "golang",        // belong to golang type
+					ModuleName: name,            // image name
+					Version:    version,         // tag
+					FileName:   scan.Kubernetes, // high-level k8s scan
+				}
+				appPkgs = append(appPkgs, k8sAppRes)
```

Comment:
Could you create a `map[string]bool` containing one element, so we can use `somemap[name]` to determine if it's in the list?  For example, https://go.dev/play/p/pxdDr5zptHM